File: contacting-conf-web/src/main/java/nl/ing/api/contacting/conf/domain/model/surveysetting/SurveyPhoneNumberFormatVO.java
package nl.ing.api.contacting.conf.domain.model.surveysetting;

import nl.ing.api.contacting.conf.domain.validation.SurverySettingValidation;

import java.util.Optional;

public record SurveyPhoneNumberFormatVO(
        Optional<Long> id,
        long surveyId,
        String format,
        String direction
) {
    public SurveyPhoneNumberFormatVO {
        SurverySettingValidation.validateId(id);
        SurverySettingValidation.validateSurveyId(surveyId);
        SurverySettingValidation.validateFormat(format);
        SurverySettingValidation.validateDirection(direction);
    }
}

File: contacting-conf-web/src/main/scala/nl/ing/api/contacting/conf/domain/SurveyPhoneNumberFormatVO.scala
package nl.ing.api.contacting.conf.domain

import nl.ing.api.contacting.conf.domain.types.ContactingTypes.DatabaseId
import nl.ing.api.contacting.conf.domain.types.ContactingTypes.SurveyPhNumDirection

case class SurveyPhoneNumberFormatVO(id: Option[DatabaseId], surveyId: DatabaseId, format: String, direction: SurveyPhNumDirection)
